### PR TITLE
Sonar: Invoke method(s) only conditionally

### PR DESCRIPTION
**Pull request by qmonmert**
Fix https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&id=jhipster-sample-application&open=AYJOgZvrN-MmhQ9YSMdt 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs [hunk 1]
```diff
@@ -26,7 +26,9 @@
 _%>
 <%_ if (jpaMetamodelFiltering) { _%>
     public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(<%= entityClass %>Criteria criteria<% if (!paginationNo) { %>, @org.springdoc.api.annotations.ParameterObject Pageable pageable<% if (reactive) { %>, ServerHttpRequest request<% } %><% } %>) {
+      if (log.isDebugEnabled()) {  
         log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+      }
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs:31]:
```suggestion
        log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"));
```
I find the supplier solution more elegant 😄. What do you think

File: generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs [hunk 2]
```diff
@@ -95,7 +97,9 @@ public class <%= serviceClassName %> extends QueryService<<%= persistClass %>> {
      */
     @Transactional(readOnly = true)
     public Page<<%= instanceType %>> findByCriteria(<%= criteria %> criteria, Pageable page) {
-        log.debug("find by criteria : {}, page: {}", criteria.toString().replaceAll("[\n\r\t]", "_"), page);
+        if (log.isDebugEnabled()) {
+            log.debug("find by criteria : {}, page: {}", criteria.toString().replaceAll("[\n\r\t]", "_"), page);
+        }
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs:102]: page is missing the replacement too.
https://sonarcloud.io/project/issues?resolved=false&types=VULNERABILITY&id=jhipster-sample-application&open=AXk8UlfKmxD5nLr49ZDM

**Review by mshima — COMMENTED** [review]
@qmonmert your are fast.
I didn't had time to fix it this weekend.

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs:31]: yes 👍

**Comment by mshima** [comment]
Opened an issue at prettier-java https://github.com/jhipster/prettier-java/issues/545

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs [hunk 3]
```diff
@@ -26,7 +26,7 @@
 _%>
 <%_ if (jpaMetamodelFiltering) { _%>
     public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(<%= entityClass %>Criteria criteria<% if (!paginationNo) { %>, @org.springdoc.api.annotations.ParameterObject Pageable pageable<% if (reactive) { %>, ServerHttpRequest request<% } %><% } %>) {
-        log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+      log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"));
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs:29]:
```suggestion
      log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs [hunk 4]
```diff
@@ -45,7 +45,7 @@ _%>
      */
     @GetMapping("/<%= entityApiUrl %>/count")
     public ResponseEntity<Long> count<%= entityClassPlural %>(<%= entityClass %>Criteria criteria) {
-        log.debug("REST request to count <%= entityClassPlural %> by criteria: {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+        log.debug("REST request to count <%= entityClassPlural %> by criteria: {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"));
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs:48]:
```suggestion
        log.debug("REST request to count <%= entityClassPlural %> by criteria: {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```

File: generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs [hunk 5]
```diff
@@ -78,7 +78,7 @@ public class <%= serviceClassName %> extends QueryService<<%= persistClass %>> {
      */
     @Transactional(readOnly = true)
     public List<<%= instanceType %>> findByCriteria(<%= criteria %> criteria) {
-        log.debug("find by criteria : {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+        log.debug("find by criteria : {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"));
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs:81]:
```suggestion
        log.debug("find by criteria : {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```

File: generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs [hunk 6]
```diff
@@ -95,7 +95,7 @@ public class <%= serviceClassName %> extends QueryService<<%= persistClass %>> {
      */
     @Transactional(readOnly = true)
     public Page<<%= instanceType %>> findByCriteria(<%= criteria %> criteria, Pageable page) {
-        log.debug("find by criteria : {}, page: {}", criteria.toString().replaceAll("[\n\r\t]", "_"), page);
+        log.debug("find by criteria : {}, page: {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"), page);
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs:98]:
```suggestion
        log.debug("find by criteria : {}, page: {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"), () -> page.toString().replaceAll("[\n\r\t]", "_"));
```

File: generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs [hunk 7]
```diff
@@ -112,7 +112,7 @@ public class <%= serviceClassName %> extends QueryService<<%= persistClass %>> {
      */
     @Transactional(readOnly = true)
     public long countByCriteria(<%= criteria %> criteria) {
-        log.debug("count by criteria : {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+        log.debug("count by criteria : {}", () => criteria.toString().replaceAll("[\n\r\t]", "_"));
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/service/EntityQueryService.java.ejs:115]:
```suggestion
        log.debug("count by criteria : {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```

File: generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs [hunk 8]
```diff
@@ -26,7 +26,7 @@
 _%>
 <%_ if (jpaMetamodelFiltering) { _%>
     public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(<%= entityClass %>Criteria criteria<% if (!paginationNo) { %>, @org.springdoc.api.annotations.ParameterObject Pageable pageable<% if (reactive) { %>, ServerHttpRequest request<% } %><% } %>) {
-        log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", criteria.toString().replaceAll("[\n\r\t]", "_"));
+      log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```
Comment (mshima) [inline generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs:29]:
```suggestion
        log.debug("REST request to get <%= entityClassPlural %> by criteria: {}", () -> criteria.toString().replaceAll("[\n\r\t]", "_"));
```

**Comment by mshima** [comment]
Slf4j logger doesn't support supplier support 😞.

**Pull request closed without merging**